src/dar/DarFileStorage.js

```diff
@@ -83,6 +83,8 @@ export default class DarFileStorage {
     let { dir, name } = path.parse(darpath)
     // ATTENTION: it is probably possible to create collisions here if somebody uses '@' in a bad way
     // for now, I accepting this because I don't think that this is realistic
+    // dir folder must end with slash since later we will build path again
+    dir += '/'
```

Comment:
It would be better to check if the last character is slash, and only insert if not present, otherwise we could have double slashes

Reply:
Or is it so that path.parse() results in without '/' ? If so, then we should add this to the comment.

Reply:
I checked this. The API says that this is never with '/' so this could should be fine.

Reply:
I have updated the comment accordingly.